ClientLib/src/main/java/com/o3dr/services/android/lib/drone/companion/solo/tlv/SoloPanoOptions.java

```diff
@@ -0,0 +1,157 @@
+package com.o3dr.services.android.lib.drone.companion.solo.tlv;
+
+import android.os.Parcel;
+import android.support.annotation.IntDef;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Created by phu on 6/24/16.
+ */
+public class SoloPanoOptions extends SoloShotOptions {
+
+    public static final int MESSAGE_LENGTH = 12;
+
+    /**
+     * Pano can automatically run, is it running?
+     */
+    private static final int PANO_ON_VALUE = 1;
+    private static final int PANO_OFF_VALUE = 0;
+
+    private boolean isRunning;
+
+    /**
+     * Pano has multiple sub modes
+     */
+    @IntDef({
+            PANO_PREFERENCE_CYLINDRICAL,
+            PANO_PREFERENCE_SPHERICAL,
+            PANO_PREFERENCE_VIDEO
+    })
+    public @interface PanoPreference{}
+
+    public static final int PANO_PREFERENCE_CYLINDRICAL = 0;
+    public static final int PANO_PREFERENCE_SPHERICAL = 1;
+    public static final int PANO_PREFERENCE_VIDEO = 2;
+    @PanoPreference
+    private int panoPreference;
+
+    /**
+     * Pan angle (used in cylindrical)
+     */
+    private short panAngle;
+
+    /**
+     * Yaw speed (used in video)
+     */
+    private float degreesPerSecondYawSpeed;
+
+    /**
+     * Camera FOV
+     */
+    private float cameraFOV;
+
+    public boolean isRunning() {
+        return isRunning;
+    }
+
+    public void setRunning(boolean running) {
+        isRunning = running;
+    }
+
+    @PanoPreference
+    public int getPanoPreference() {
+        return panoPreference;
+    }
+
+    public void setPanoPreference(int panoPreference) {
```

Comment:
@phmagic add the `@PanoPreference` annotation to the method parameter to restrict the argument that can be passed to it.